applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.h

```diff
@@ -202,6 +206,125 @@ class PotentialWallCondition : public Condition
         return pNewCondition;
     }
 
+    /// Find the condition's parent element.
+	void Initialize() override
+	{
+        KRATOS_TRY;
+
+        if(this->Id()==1)
+        {
+            std::cout << "Condition = " << this->Id()  << std::endl;
+            std::cout << "mInitializeWasPerformed = " << mInitializeWasPerformed  << std::endl;
+        }
+        
+        //std::cout << "Condition = " << this->Id()  << std::endl;
+
+		const array_1d<double,3>& rNormal = this->GetValue(NORMAL);
+		if (norm_2(rNormal) == 0.0)
+		  {
+		    std::cout << "error on condition -> " << this->Id() << std::endl;
+		    KRATOS_THROW_ERROR(std::logic_error, "NORMAL must be calculated before using this condition","");
+		  }
+
+		if (mInitializeWasPerformed)
+		{
+            std::cout << "Returning early"  << std::endl;
+            std::cout << "Condition = " << this->Id()  << std::endl;
+			return;
+		}
+
+		mInitializeWasPerformed = true;
+
+		double EdgeLength;
+		array_1d<double,3> Edge;
+		GeometryType& rGeom = this->GetGeometry();
+		WeakPointerVector<Element> ElementCandidates;
+		for (SizeType i = 0; i < TDim; i++)
+		{
+			WeakPointerVector<Element>& rNodeElementCandidates = rGeom[i].GetValue(NEIGHBOUR_ELEMENTS);
+			for (SizeType j = 0; j < rNodeElementCandidates.size(); j++)
+			{
+				ElementCandidates.push_back(rNodeElementCandidates(j));
+			}
+		}
+
+		std::vector<IndexType> NodeIds(TNumNodes), ElementNodeIds;
+
+		for (SizeType i=0; i < TNumNodes; i++)
+		{
+			NodeIds[i] = rGeom[i].Id();
+		}
+
+        std::sort(NodeIds.begin(), NodeIds.end());
+        
+        if(this->Id()==1)
+        {
+            std::cout << "Condition = " << this->Id()  << std::endl;
+            std::cout << "ElementCandidates.size() = " << ElementCandidates.size()  << std::endl;
+        }
+
+		for (SizeType i=0; i < ElementCandidates.size(); i++)
+		{
+			GeometryType& rElemGeom = ElementCandidates[i].GetGeometry();
+			ElementNodeIds.resize(rElemGeom.PointsNumber());
+
+			for (SizeType j=0; j < rElemGeom.PointsNumber(); j++)
```

Comment:
Reuse `GetSortedIds(GeometryType&)`

Reply:
I dont think I can reuse GetSortedIds because NodesIds and ElementNodeIds have different sizes.

Reply:
I therefore made a new function, but I can change it if we find the way of reusing the previous one.

Reply:
It will work because `GetSortedIds` sets the size according to the geometry you pass it. For the condition geometry this will be e.g. 2 and for the element geometry 3.